src/helpers/storage.js

```diff
@@ -1,34 +1,135 @@
+  return JSON.parse(localStorage.getItem(STORAGE_TASK_KEY)) || [];
 };
 
 export const addTaskToStorage = (taskTitle) => {
   const currentTasks = getTasksDataFromLocalStorage();
   const newTasks = [...currentTasks, taskTitle];
-  saveDataInStorageForKey(STORAGE_KEY, newTasks);
+  saveDataInStorageForKey(STORAGE_TASK_KEY, newTasks);
 };
 
 export const deleteTaskStorage = (id) => {
   const currentTasks = getTasksDataFromLocalStorage();
   const indexOfDeletedTask = currentTasks.findIndex((task) => task.id === id);
   const newTaskList = currentTasks.filter((task) => task !== currentTasks[indexOfDeletedTask]);
-  saveDataInStorageForKey(STORAGE_KEY, newTaskList);
+  saveDataInStorageForKey(STORAGE_TASK_KEY, newTaskList);
 };
 
 export const updateTaskInStorage = (task) => {
   const currentTasks = getTasksDataFromLocalStorage();
-  const updatedTasks = currentTasks.map((currentTask) => {
-    if (currentTask.id === task.id) {
-      return task;
+  const updatedTasks = currentTasks.map((currentTask) =>
+    currentTask.id === task.id ? task : currentTask,
+  );
+  saveDataInStorageForKey(STORAGE_TASK_KEY, updatedTasks);
+};
+
+export const getGroupOfTasksDataFromLocalStorage = () => {
+  return JSON.parse(localStorage.getItem(STORAGE_GROUP_OF_TASKS_KEY)) || [];
+};
+
+// THIS IS ANOTHER SOLUTION OF updateTaskInGroupStorage LEFT BY PURPOSE
+
+// export const updateTaskInGroupStorage = (task) => {
+//   const currentGroupOfTasks = getGroupOfTasksDataFromLocalStorage();
+//   console.log(currentGroupOfTasks, task);
+//   const foundTaskGroup = currentGroupOfTasks.find((group) => {
+//     return group.tasks.find((item) => item.id === task.id);
+//   });
+//   const newUpdatedTasksInGroup = foundTaskGroup.tasks.map((item) =>
+//     item.id === task.id ? task : item,
+//   );
+//   const newUpdatedGroup = {
+//     ...foundTaskGroup,
+//     tasks: newUpdatedTasksInGroup,
+//   };
+
+//   const updatedAllGroups = currentGroupOfTasks.map((groupItem) =>
+//     groupItem.id === newUpdatedGroup.id ? newUpdatedGroup : groupItem,
+//   );
+//   saveDataInStorageForKey(STORAGE_GROUP_OF_TASKS_KEY, updatedAllGroups);
+// };
+
+export const updateTaskInGroupStorage = (task) => {
+  const currentGroupOfTasks = getGroupOfTasksDataFromLocalStorage();
+  const updatedAllGroups = [...currentGroupOfTasks].map((group) => {
+    if (group.id === task.groupId) {
+      return {
+        ...group,
+        tasks: group.tasks.map((taskItem) => (taskItem.id === task.id ? task : taskItem)),
+      };
+    }
+    return group;
+  });
+
+  saveDataInStorageForKey(STORAGE_GROUP_OF_TASKS_KEY, updatedAllGroups);
+};
+
+export const deleteTaskInGroupStorage = (task, id) => {
+  const currentGroupOfTasks = getGroupOfTasksDataFromLocalStorage();
+  const foundGroupOfTasks = [...currentGroupOfTasks].find((group) => group.id === task.groupId);
+  const indexOfDeletedTask = foundGroupOfTasks.tasks.findIndex((tasks) => tasks.id === id);
```

Comment:
What youre doing here is finding index of task inside array to delete, but you already have id of task to delete. Only thing you need check is in line 79 - if ID of currentlyFilteredElement !== ID of element to delete. By doing what you did you have one extra layer of logic (for loops/maps/filters) on big sets of data are expensive operations what influence performance. You searched for information you actually do not need.

Reply:
done